### PR TITLE
Add pytest marker to mark tests requiring IDAES solvers

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
## Fixes/Addresses:

(replace this with the issue # fixed or addressed, if no issue exists then a brief statement of what this PR does)

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: watertap/examples/flowsheets/full_treatment_train/analysis/tests/test_flowsheets.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
                                                                flowsheet_softening,
                                                                flowsheet_softening_two_stage)
 
+
```
Comment (bknueven) [inline watertap/examples/flowsheets/full_treatment_train/analysis/tests/test_flowsheets.py]:
```suggestion
```

File: watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
 from pyomo.environ import TransformationFactory
 from pyomo.network import Arc
 
+
```
Comment (bknueven) [inline watertap/examples/flowsheets/full_treatment_train/flowsheet_components/chemistry/tests/test_posttreatment.py]:
```suggestion
```

File: watertap/unit_models/tests/test_reverse_osmosis_1D.py [hunk 3]
```diff
@@ -336,6 +336,7 @@ def test_calculate_scaling(self, RO_frame):
     def test_initialize(self, RO_frame):
         initialization_tester(RO_frame, fail_on_warning=True)
 
+    @pytest.mark.requires_idaes_solver
```
Comment (bknueven) [inline watertap/unit_models/tests/test_reverse_osmosis_1D.py]:
```suggestion
```

**Review by bknueven — COMMENTED** [review]
Some minor cleanup ...

**Comment by bknueven** [comment]
@lbianchi-lbl I made some changes to merge in #328 and #336 cleanly.

While we don't need this marker in `main` anymore, it's now needed for #319, so this should be merged soon.

**Review by bknueven — APPROVED** [review]
This is a very helpful addition with our experimental macOS support.

We should ignore the patch coverage failure -- this will get coverage again once it's used in #319.

**Comment by lbianchi-lbl** [comment]
@bknueven Thanks for the clarification and implementing the changes. Since this is needed for #319, why don't I go ahead and merge this in. If any issue pops up, we can deal with it there.

**Pull request merged**